src/core/search/compressed_sorted_set.cc

```diff
@@ -109,6 +95,15 @@ CompressedSortedSet::EntryLocation CompressedSortedSet::LowerBound(IntType value
 // to be stored to encode the triple A V B. Those are stored where diff0 = B - A was previously
 // stored, possibly extending the vector
 void CompressedSortedSet::Insert(IntType value) {
+  if (tail_value_ && *tail_value_ == value)
+    return;
+
+  if (tail_value_ && value > *tail_value_) {
+    PushBackDiff(value - *tail_value_);
+    tail_value_ = value;
```

Comment:
Perhaps it would be better to move the `tail_value_` assignment into `PushBackDiff()`

Reply:
PushBackDiff doesn't have access to the tail value / last value and I don't really wanna overload it with it